packages/edit-site/src/store/actions.js

```diff
@@ -260,9 +246,11 @@ export const setPage =
 			page.path = getPathAndQueryString( entity?.link );
 		}
 
-		const template = await registry
-			.resolveSelect( coreStore )
-			.__experimentalGetTemplateForLink( page.path );
+		const { data: template } = await apiFetch( {
```

Comment:
Did you do that because of the caching?

Reply:
What about tweaking `__experimentalGetTemplateForLink.shouldInvalidate` to invalidate the cache at the right moment?

Reply:
Yes(see [here](https://github.com/WordPress/gutenberg/pull/51477#discussion_r1235077297)). Could we invalidate the cache? I tried it a bit but couldn't..

Reply:
That's great! I pushed a change for that. It's a shame that my first tries were to do just that, and because I didn't find a way to do it then, I circled around the proper solution for so long..